docs/source/reference.rst

```diff
@@ -107,18 +107,39 @@ Semantics
 
 This library generally tries hard to match the semantics of Python
 3.6's native async generators in every detail (`PEP 525
-<https://www.python.org/dev/peps/pep-0525/>`__), except that it adds
-``yield from`` support, and it doesn't currently support the
-``sys.{get,set}_asyncgen_hooks`` garbage collection API. There are two
-main reasons for this: (a) it doesn't exist on Python 3.5, and (b)
-even on 3.6, only built-in generators are supposed to use that API,
-and that's not us. In any case, you probably shouldn't be relying on
-garbage collection for async generators – see `this discussion
+<https://www.python.org/dev/peps/pep-0525/>`__), with additional
+support for ``yield from`` and for returning non-None values from
+an async generator (under the theory that these may well be added
+to native async generators one day).
+
+
+Garbage collection hooks
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This library fully supports the native async generator
+`finalization semantics <https://www.python.org/dev/peps/pep-0525/#finalization>`__,
+including the per-thread ``firstiter`` and ``finalizer`` hooks.
+You can use ``async_generator.set_asyncgen_hooks()`` exactly
+like you would use ``sys.set_asyncgen_hooks()`` with native
+generators. On Python 3.6+, the former is an alias for the latter,
+so libraries that use the native mechanism should work seamlessly
+with ``@async_generator`` functions.
```

Comment:
We should probably include a warning here that not all async libraries will *use* these hooks. For example presumably trio will use these if it uses any, but if you're using asyncio on 3.5 then it won't do the same async generator cleanup that it does on 3.6+.

Reply:
good thought - I added a sentence about this. (When we teach trio about these, we can go back and say "but if you're using trio, it will just work!" if we want. :P)